fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java

```diff
@@ -728,47 +712,12 @@ private void initAggregateBuiltins() {
                 continue; // Only function `Count` support pseudo types now.
             }
 
-            // count in multi distinct
-            if (t.isChar() || t.isVarchar()) {
-                addBuiltin(AggregateFunction.createBuiltin(FunctionSet.MULTI_DISTINCT_COUNT, Lists.newArrayList(t),
-                        Type.BIGINT,
-                        Type.VARCHAR,
-                        false, true, true));
-            } else if (t.isBoolean() || t.isTinyint() || t.isSmallint() || t.isInt() || t.isBigint() ||
-                    t.isLargeint() || t.isFloat() || t.isDouble()) {
-                addBuiltin(AggregateFunction.createBuiltin(FunctionSet.MULTI_DISTINCT_COUNT, Lists.newArrayList(t),
-                        Type.BIGINT,
-                        Type.VARCHAR,
-                        false, true, true));
-            } else if (t.isDate() || t.isDatetime()) {
-                addBuiltin(AggregateFunction.createBuiltin(FunctionSet.MULTI_DISTINCT_COUNT, Lists.newArrayList(t),
-                        Type.BIGINT,
-                        Type.VARCHAR,
-                        false, true, true));
-            } else if (t.isDecimalV2() || t.isDecimalV3()) {
-                addBuiltin(AggregateFunction.createBuiltin(FunctionSet.MULTI_DISTINCT_COUNT, Lists.newArrayList(t),
-                        Type.BIGINT,
-                        Type.VARCHAR,
-                        false, true, true));
-            }
+            // MULTI_DISTINCT_COUNTM
+            addBuiltin(AggregateFunction.createBuiltin(FunctionSet.MULTI_DISTINCT_COUNT, Lists.newArrayList(t),
```

Comment:
Maybe some complex types don't support `MULTI_DISTINCT_COUNT`, such as JSON/Map/Struct.

Reply:
I'm afraid not. You need to implement it at BE at first, and each type need to define a `hash` function.

Reply:
Therefore, we should judge whether the type `t` supports `MULTI_DISTINCT_COUNT` here.

Reply:
Yep, my mistake. I doubt why MULTI_COUNT_DISTINCT is different from `COUNT_DISTINCT`. I will revert to original policy.